gc/mmtk/mmtk.c

```diff
@@ -667,7 +658,7 @@ rb_gc_impl_new_obj(void *objspace_ptr, void *cache_ptr, VALUE klass, VALUE flags
     }
 
     if (objspace->gc_stress) {
-        mmtk_handle_user_collection_request(ractor_cache);
+        mmtk_handle_user_collection_request(ractor_cache, false, false);
```

Comment:
Is there a reason we set `force` to false here?

Reply:
I observed the behavior of the default GC in the following snippet:

```ruby
puts "Hello"
GC.disable
GC.stress=true
puts GC.count
puts "Goodbye"
puts GC.count
```

The program prints the same number for both `GC.count`.  This means if both `GC.disable` and `GC.stress=true` are executed, it will not trigger GC at allocation time.  This also means `GC.disable` has higher priority over `GC.stress`.

So when using MMTk, I let it mimic the behavior or the default GC by not forcing a GC at this call site.

This behavior is not documented in the official documentation of [`GC.stress=`](https://docs.ruby-lang.org/en/master/GC.html#method-c-stress-3D) or [`GC.disable`](https://docs.ruby-lang.org/en/master/GC.html#method-c-disable).